operator/controller/src/main/java/io/apicurio/registry/operator/resource/app/AppDeploymentResource.java

```diff
@@ -99,15 +100,40 @@ public static void addEnvVar(Map<String, EnvVar> map, EnvVar envVar) {
         }
     }
 
+    /**
+     * Get container with a given name from the given Deployment.
+     *
+     * @throws OperatorException if container was not found
+     */
     public static Container getContainer(Deployment d, String name) {
         requireNonNull(d);
         requireNonNull(name);
-        for (var c : d.getSpec().getTemplate().getSpec().getContainers()) {
-            if (name.equals(c.getName())) {
+        log.debug("Getting container {} in Deployment {}", name, ResourceID.fromResource(d));
+        if (d.getSpec() != null & d.getSpec().getTemplate() != null) {
+            var c = getContainer(d.getSpec().getTemplate(), name);
+            if (c != null) {
                 return c;
             }
         }
         throw new OperatorException(
                 "Container %s not found in Deployment %s".formatted(name, ResourceID.fromResource(d)));
     }
+
+    /**
+     * Get container with a given name from the given PTS.
+     *
+     * @return null when container was not found
+     */
+    public static Container getContainer(PodTemplateSpec pts, String name) {
```

Comment:
rename to `getUserDefinedContainer` ?
I understand why you need the two different implementations and that's fine, but the overload force the reader to always check the types.

Reply:
I can rename that, but this is really nitpicking. The methods do the same thing but take different first argument, which should be easily distinguishable. And there is also an IDE.

Reply:
> but this is really nitpicking

Let me elaborate on this, 2 methods with different semantics named the same is arguably confusing as you need to check the types with the IDE, while, just changing the name, you can review in GH with simple syntax highlighting.

Reply:
Done.